plugins/registry.js

```diff
@@ -49,6 +49,39 @@ const resolveReactElement = (element) => {
     return element;
 };
 
+class SubMenuGroup {
+    constructor(pluginId, parentId, depth) {
+        this.pluginId = pluginId;
+        this.parentId = parentId;
+        this.depth = depth;
+        this.registerMenuItem = this.registerMenuItem.bind(this);
+        this.subMenuGroup = [];
+    }
+    registerMenuItem(item, action, filter) {
+        this.id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_MENUITEM,
```

Comment:
Instead of evaluating `action.text.subMenu`, `action.data.parentId` was chosen to determine if the element is a submenu.  I did this because the root of the menu was already dispatching `ActionTypes.RECEIVED_PLUGIN_COMPONENT`, and it felt cleaner in the reducer.  

Having a submenu item treated exactly like a menu item without a submenu would not work in the current setup, as the reducer needs to find/add the sub menu item in the tree to save it into state.